engine/src/opensslsocket.cpp

```diff
@@ -1198,6 +1198,13 @@ MCSocket *MCS_accept(uint2 port, MCObject *object, MCNameRef message, Boolean da
 		listen(sock, SOMAXCONN);
 #endif
     
+    if (0 == port)
+    {
+        socklen_t addrsize = sizeof(addr);
+        getsockname(sock, (sockaddr *)&addr, &addrsize);
```

Comment:
yes OK, in the event of failure here I guess we should close, set a result error and return NULL.